components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -580,6 +614,266 @@ protected void initiateAuthenticationRequest(HttpServletRequest request, HttpSer
+                filteredScopes.append(scopes[1]).append(SPACE);
+            }
+        }
+
+        String requestedScopes = filteredScopes.toString();
+        if (LOG.isDebugEnabled() && StringUtils.isBlank(requestedScopes)) {
+            LOG.debug("No valid values found for the authenticator " + authenticatorName + " in the query parameter " +
+                    OIDCAuthenticatorConstants.FEDERATED_TOKEN_SCOPE + " for federated token sharing, IDP: " +
+                    authenticatorName);
+        }
+
+        return filteredScopes.toString();
+    }
+
+    /**
+     * This method evaluates whether application has requested to share the token. The first priority is given to the
+     * authenticator parameters set at the adaptive script. Then the query parameters.
+     *
+     * @param context The authentication context.
+     * @return Whether the application has requested to share the token.
+     */
+    private boolean requestedToShareFederatedToken(AuthenticationContext context) {
+
+        String authenticatorName = getFederatedAuthenticatorName(context);
+        // The first priority is given to the parameters setup at the adaptive script. Then the query parameters.
+        String shareFederatedToken =
+                getAdaptiveScriptValues(context, OIDCAuthenticatorConstants.SHARE_FEDERATED_TOKEN_PARAM);
+
+        if (LOG.isDebugEnabled() && StringUtils.isNotBlank(shareFederatedToken)) {
+            LOG.debug("Adaptive script parameter " + OIDCAuthenticatorConstants.SHARE_FEDERATED_TOKEN_PARAM +
+                    " found for federated token sharing, IDP: " + authenticatorName);
+        }
+        // Checks if the token sharing is requested via adaptive script.
+        if (StringUtils.isBlank(shareFederatedToken)) {
+            // Checks if the token sharing is requested via authorize request query parameters.
+            shareFederatedToken = getQueryParameter(context, OIDCAuthenticatorConstants.SHARE_FEDERATED_TOKEN_PARAM);
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("No adaptive script parameter: " + OIDCAuthenticatorConstants.SHARE_FEDERATED_TOKEN_PARAM +
+                        " found. Query parameter: " + OIDCAuthenticatorConstants.SHARE_FEDERATED_TOKEN_PARAM +
+                        " value: " + shareFederatedToken + " found for federated token sharing, IDP: "
+                        + authenticatorName);
+            }
+        }
+        return Boolean.parseBoolean(shareFederatedToken);
+    }
+
+    /**
+     * This method is used to retrieve the query parameters from the authentication request.
+     *
+     * @param context        The authentication context with authentication request.
+     * @param queryParamName The required query parameter name.
+     * @return The query parameter value.
+     */
+    private String getQueryParameter(AuthenticationContext context, String queryParamName) {
+
+        String authenticatorName = getFederatedAuthenticatorName(context);
```

Comment:
this is only used in the debug logs, lets use this directly.

Reply:
Fixed d2b40f9255b4a5ef3b8f9d44bd58ccdba9f50f70